commons/src/tracing.rs

```diff
@@ -0,0 +1,107 @@
+//! Tracing service.
+
+use opentelemetry::api::{
+    Carrier, HttpTextFormat, Key, Provider, Span, SpanContext, TraceContextPropagator,
+};
+use opentelemetry::{global, sdk};
+use opentelemetry_jaeger::{Exporter, Process};
+use std::collections::HashMap;
+
+use actix_web::dev::ServiceRequest;
+use actix_web::http;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+
+/// init_tracer sets up Jaeger tracer
+pub fn init_tracer(name: &'static str, maybe_agent_endpoint: Option<String>) -> thrift::Result<()> {
```

Comment:
What's the point of using `thrift::Result` here?

Reply:
`opentelemetry_jaeger::Builder.init()` may thrown an error, compatible with `Result`, so `init_tracer` caller has to handle this error

Reply:
I think it'd be cleaner if we'd convert it to our error type. The `?` operator is likely to enough to do so.

Reply:
Good idea, implemented in 348aa84